src/rgw/rgw_rest_user_policy.cc

```diff
@@ -131,6 +132,14 @@ void RGWPutUserPolicy::execute()
     return;
   }
 
+  if (!store->svc()->zone->is_meta_master()) {
+    ceph::bufferlist in_data;
+    op_ret = forward_request_to_master(s, nullptr, store, in_data, nullptr);
+    if (op_ret < 0) {
+      ldpp_dout(this, 0) << "forward_request_to_master returned ret=" << op_ret << dendl;
+    }
```

Comment:
In case of error on master zone, do we still go ahead and create the policy in the other zone?

Reply:
sorry this code wasn't supposed to be here, I hadn't pushed the latest changes

Reply:
updated, for PUT requests, failure in master doesn't proceed for PUT in the local zone, the only exception is in delete requests where master returns a -ENOENT